compiler/src/main/scala/org/scalajs/nscplugin/GenJSCode.scala

```diff
@@ -6605,6 +6605,17 @@ abstract class GenJSCode[G <: Global with Singleton](val global: G)
   def isStaticModule(sym: Symbol): Boolean =
     sym.isModuleClass && !isImplClass(sym) && !sym.isLifted
 
+  def isAnonymousClass(sym: Symbol): Boolean = {
+    /* sym.isAnonymousClass simply checks if
+     * `name containsName tpnme.ANON_CLASS_NAME`
+     * so after flatten (which we are) it identifies classes nested inside
+     * anonymous classes as anonymous (notably local classes, see #4278).
+     *
+     * Instead we apply the check on the sym's originalName.
+     */
+    sym.originalName.containsName(tpnme.ANON_CLASS_NAME)
```

Comment:
Ach, it's supposed to be `unexpandedName` (`originalName` is deprecated since 2.11.0):
```suggestion
    sym.unexpandedName.containsName(tpnme.ANON_CLASS_NAME)
```